client/src/data.js

```diff
@@ -0,0 +1,86 @@
+export async function readListing() {
+  const req = {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
```

Comment:
I suggest for your final project that you use localStorage instead of sessionStorage. All you need to do is change the name from `session` to `local` everywhere that you use it.